fixing issue related with https with self signed certs

carvilsi opened the pull request.

gvdent commented on src/SocketController.coffee:
```diff
@@ -11,6 +11,7 @@ class SocketController
 
     @options = @options or defaultOptions
     @options.query = "sdkVersion=#{pjson.version}&requiredServerVersion=#{pjson.com_weaverplatform.requiredServerVersion}"
+    @options.rejectUnauthorized = false
```
This option should not be set to false in the default SDK, since it allows for MITM attacks. Maybe a separate release for those who like to live on the edge (or self sign certificates).

carvilsi replied:
I propose to add a config var to set this value to true/false.

Gielert replied:
Yes, this would be a better option

carvilsi commented:
Added at config: `rejectUnauthorized` and set to *true* by default.
Added `WEAVER_REJECT_UNAUTHORIZED` as environment variable
Small require cleaning on File test.

gvdent commented on dist/weaver-sdk.full.js:
```diff
@@ -99666,6 +99666,7 @@ module.exports={
       };
       this.options = this.options || defaultOptions;
       this.options.query = "sdkVersion=" + pjson.version + "&requiredServerVersion=" + pjson.com_weaverplatform.requiredServerVersion;
+      this.options.rejectUnauthorized = false;
```
I think you need to run npm-prepublish one more time, since this differs from the coffee code.

gvdent commented on src/SocketController.coffee:
```diff
@@ -11,6 +12,7 @@ class SocketController
 
     @options = @options or defaultOptions
     @options.query = "sdkVersion=#{pjson.version}&requiredServerVersion=#{pjson.com_weaverplatform.requiredServerVersion}"
+    @options.rejectUnauthorized = config.get('weaver.rejectUnauthorized')
```
👍
